tools/check_spelling.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Applies requisite code formatters to the source tree
+# check_spelling.sh
+
+set -u
+set -e
+
+VERSION="0.3.4"
+linux_misspell_sha="34d489dbc5ddb4dfd6d3cfac9fde8660e6c37e6c"
+mac_misspell_sha="f2607e2297b9e8af562e384c38045033375c7433"
+
+OS=""
+
+MISSPELL_ARGS="-error -o stderr"
+
+if [[ $# -lt 1 ]]; then
+  echo "Usage: $0 check|fix"
+  exit -1
+fi
+
+if [[ $1 == "fix" ]]; then
+  MISSPELL_ARGS="-w"
+fi
+
+if [[ `uname` == "Darwin" ]]; then
+  OS="mac"
+elif [[ `uname` == "Linux" ]]; then
+  OS="linux"
+else
+  echo "Current only support mac/Linux"
+  exit 1
+fi
+
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+ROOTDIR=$SCRIPTPATH/..
+cd "$ROOTDIR"
+
+BIN_FILENAME="misspell_${VERSION}_${OS}_64bit.tar.gz"
+# Install tools we need
+if [[ ! -e "/tmp/misspell" ]]; then
+  if ! wget https://github.com/client9/misspell/releases/download/v${VERSION}/${BIN_FILENAME} -O /tmp/${BIN_FILENAME} --no-verbose --tries=3 -o /tmp/wget.log; then
+    cat /tmp/wget.log
```

Comment:
Ideally we could specify `$TMPDIR` in the environment here rather than hardcoding.